base/strings/unicode.jl

```diff
@@ -385,7 +386,14 @@ julia> islowercase('❤')
 false
 ```
 """
-islowercase(c::AbstractChar) = ismalformed(c) ? false : Bool(ccall(:utf8proc_islower, Cint, (UInt32,), UInt32(c)))
+function islowercase(c::AbstractChar)
+    if ismalformed(c)
+        false
+    else
+        ui = UInt32(c)::UInt32 # type-assertion to ensure that we may assume :foldable
+        @assume_effects :foldable Bool(ccall(:utf8proc_islower, Cint, (UInt32,), ui))
+    end
+end
```

Comment:
```suggestion
islowercase(c::AbstractChar) = ismalformed(c) ? false :
    Bool(@assume_effects :foldable @ccall utf8proc_islower(UInt32(c)::UInt32)::Cint)
```
This version directly applies to `@ccall` and it doesn't assume any effects for conversions to `UInt32`, allowing for a more straightforward description.